### DISCUSSION TITLE
Help launching Freeplane portable 1.10.2 with Java issues

### DISCUSSION THREAD

**Discussion by quickfold** [freeplane/freeplane#365] [Help] [answered]
I downloaded FP 1.10.2 stable with the Java installer for Windows. I can get FP to launch, but when I try to launch my FP Portable 1.10.2, it says that it can't create the Java Virtual Machine. I tried to point FP to Java using the instructions from another post [here](https://www.happycoders.eu/java/how-to-switch-multiple-java-versions-windows/) but it didn't work. I don't have a directory `Java\jdk18` so I'm not sure where to point JAVA_HOME or if there is a better approach for setting up FP Portable 1.10.2 to launch properly. Can someone help me?

**Comment by abc16361** [comment]
I am using java-17.0.3, and just tested the portable version, and it works. After installing java-17.0.3, it needs to create JAVA_HOME on the `System Variables`  and set the value with `C:\Program Files\Java\jdk-17.0.3`, and then `reboot` Windows. I haven't touched java-18, so no comments about that. If java-18 doesn't work for you in this way, you can try java-17 to see how it works.

**Reply by abc16361** [reply]
@quickfold 
Sorry, I misunderstood what you mean. I thought you were talking about java installation, so what I suggested can't be applied to your case.

**Comment by dpolivaev** [comment, marked as answer]
install https://sourceforge.net/projects/portableapps/files/OpenJDK%20JRE%20Portable/OpenJDKJRE64_15.0.2_Build_7.paf.exe/download on your portable device or directory

**Reply by quickfold** [reply]
Thanks, that worked, but I'm confused, because I ran FP Portable before 1.10.2, and I didn't have portable Java installed. Is this build different in requiring portable Java?

**Reply by dpolivaev** [reply]
Now it needs Java 11 or later and needs to know how to find it. I think the old version was happy with Java 8

**Reply by vascotenner** [reply]
I have installed them both in the same directory (hence I have folders \freeplane and \commonfiles). However, it still gives a java error.  How to solve?

I found this workaround, but is seems tedious:
Install java portable

Set java home:
https://www.shellhacks.com/set-environment-variable-without-admin-rights-windows/
https://explainjava.com/set-java-path-and-java-home-windows/

Add java to path
>PATH=installlocation\CommonFiles\OpenJDKJRE64\bin;%PATH%

Install freeplane (not portable)

**Reply by dpolivaev** [reply]
If you use portable app you can install portable java and portable freeplane from their portable app manager
Alternatively you can just unzip freeplane (zip archive) and java somewhere and use variable FREEPLANE_JAVA_HOME to point freeplane to java

**Reply by BobLangill** [reply]
Thanks for handling this so quickly. I use the portableapps version exclusively. I had no problem installing the paf.exe your link pointed to, and things are still working after additional updates to both the portableapps JRE and freeplanePortable 1.10.3   The active community support is something I cite anytime I recommend using Freeplane for mind mapping software.

**Reply by vladimir-from-adizes** [reply]
I suggest that portable version should include Java the same fashion as Freeplane-Setup-with-Java-1.10.3.exe package does
Otherwise it is showstopper for a common user.  Very few geeks will look for forums to look for solutions if something not working.

In fact that was solution that worked for me after I spent an hour trying to figure out what is not working and how to fix it:
1) copy "runtime" folder to FreeplanePortable folder from whereever Freeplane-Setup-with-Java-1.10.3.exe is installed
2) updated last line of FreeplanePortable\App\AppInfo\Launcher\FreeplanePortable.ini to FREEPLANE_JAVA_HOME=**%PAL:AppDir%\..\runtime**

Is there strong reason why it is not like this by default?!

P.S. Here is what regular user experiences:
1. Download, install and launch FreePlanePortable
2. Error message about java leads to [java.com](https://www.java.com/en/download/) which **downloads Java 8 (!!!)** which ("if you say so") I installed
3. Error message is still there (most people quit here) AND if you are one of the few people who pay attention to details - then you notice that error message asks for Java between 11 and 17, but there is no such option on java.com which is opened by default (more people quit here)
4. If you are super persistent - you look in forums of Portable.com and FreePlane - no obvious solution for Java 11-17

Please, spare people the hustle and bundle java inside FreePlanePortable like you did for Freeplane-Setup-with-Java-1.10.3.exe!

**Reply by dpolivaev** [reply]
PortableApps sucks. The community should convince them not me.
They support tons of apps but Freeplane was never allowed to be distributed within their ecosystem.
Java 8 is out of time. Portable app makers have added newer Java packages but I could not find any way to specify them as required.
I understand that for portable app users it might be easier to have everything packaged, but is it better? Should not willing to learn be supported? Your solution works but it feels not right to me to distribute Java with Freeplane. I am an old man. Bandwidth is costly. Memory space is costly. Forums are helpful. Documentation becomes obsolete at the moment you write it but some people took up the struggle of updating it.

I started packaging Java with Apple because they excluded Java Runtime from their world.
I started packaging Java with Windows because there are too many Java versions around and configuration is not quite trivial.
Here all you need is to install the right dependency.
https://portableapps.com/development/portableapps.com_format says they even introduced a parameter for specifying required dependencies but it is not functional yet 

> RequiresPortableApp (optional) specifies that the app requires another portable app to function. If required, this should be set to the app's AppID as used in the PortableApps.com App Store. Not yet supported by the PortableApps.com Platform

The universe sucks and there is no escape.

Probably, portable app dependency should be documented in some more prominent place.

**Reply by dpolivaev** [reply]
https://docs.freeplane.org/#/getting-started/Distributions

**Reply by BobLangill** [reply]
Dimitry, supporting FreeplanePortable now is more complicated that it should be, and that sucks. Except for a minor hiccough, it's been easy for me to continue using the PortableApps platform (whatever it is) that I've been using for the last eight years.
I appreciate your dedication to keeping things going. All respect due for that.

**Comment by sifran-github** [comment]
For reference, a solution can be found here: https://github.com/freeplane/freeplane/discussions/462#discussioncomment-2988852
